dockstore-webservice/src/main/java/io/dockstore/webservice/languages/WDLHandler.java

```diff
@@ -565,14 +577,24 @@ public static boolean versionIsGreaterThanCurrentlySupported(String semVerString
                 enhanceSemanticVersionString(LATEST_SUPPORTED_WDL_VERSION)));
     }
 
+    /**
+     * Get the version from the WDL descriptor file content. If no version is found, return the default WDL version defined by the WDL spec.
+     * @param descriptorContent
+     * @return
+     */
+    public static String getLanguageVersion(String descriptorContent) {
+        return getSemanticVersionString(descriptorContent).orElse(DEFAULT_WDL_VERSION);
```

Comment:
Edge-case, if the version line is "wrong", we will assume the default (see line 586/608). Examples, which I'm not sure if Cromwell barfs on or not:

* `version` -- this one maybe OK if we fall back on default
* `version 1.0 // Version 1.0` -- seems like it should 1.0
* `version 1 0` -- forgot the dot, not sure what Cromwell does

Reply:
Cromwell actually fails on all three examples in which case I don't think we should store the version for the failed SourceFile. It would make validation easier because a descriptor source file without a version set would indicate that something is wrong with it. Ideally, we would only use the default if there is no `version` line at all...I'm going to see if I can re-work the parsing so it's more accurate